cmd/flashback/main.go

```diff
@@ -299,10 +299,24 @@ func main() {
 		workerStates := make([]nodeWorkerState, len(nodes))
 
 		for i, n := range nodes {
-			session, err := mgo.Dial(n.url)
+			dialInfo, err := mgo.ParseURL(n.url)
+			panicOnError(err)
+			session, err := mgo.DialWithInfo(dialInfo)
 			panicOnError(err)
 			session.SetSocketTimeout(time.Duration(socketTimeout))
 			defer session.Close()
+			if dialInfo.Username != "" && dialInfo.Password != "" {
```

Comment:
Just a suggestion for cleanup. If I'm following the mgo source right, this step doesn't look necessary. The credentials are stored in the session when you call DialWithInfo, and then authentication (call to Login) is done implicitly when acquireSocket is called for various ops.

https://github.com/go-mgo/mgo/blob/v2/session.go#L438
https://github.com/go-mgo/mgo/blob/9a2573d4ae52a2bf9f5b7900a50e2f8bcceeb774/session.go#L3862